components/decimal/src/provider.rs

```diff
@@ -106,3 +107,7 @@ impl Default for DecimalSymbolsV1<'static> {
         }
     }
 }
+
+impl ResourceMarker for DecimalSymbolsV1Marker {
+    const KEY: ResourceKey = key::SYMBOLS_V1;
+}
```

Comment:
i like this idea

Reply:
Yeah!  We should do something along these lines.  I like the first version better.  I would also like to get rid of the separate key constants, because you will be able to access them via `FooV1Marker::KEY`.

Reply:
Here's another interesting proposal.  We can allow only one key per struct, and for structs with multiple keys, we make them newtypes with `#[serde(transparent)]`.  They can be cast to the inner type if needed.

Reply:
Here's what I got working:

```rust
use icu_provider::prelude::*;
use std::borrow::Cow;

// We also need `yoke` in scope: <https://github.com/unicode-org/icu4x/issues/1557>
use icu_provider::yoke;

#[icu_provider::data_struct(
    "demo/foo@1",
    BarV1Marker = "demo/bar@1",
    BazV1Marker = "demo/baz@1"
)]
pub struct FooV1<'data> {
    message: Cow<'data, str>,
};

assert_eq!(FooV1Marker::KEY.get_path(), "demo/foo@1");
assert_eq!(BarV1Marker::KEY.get_path(), "demo/bar@1");
assert_eq!(BazV1Marker::KEY.get_path(), "demo/baz@1");
```

Reply:
Cool!

Do we need custom marker names? We could derive them from the key path: `"demo/foo@1"` -> `DemoFooV1Marker` (or maybe ignore the name space as the rust module takes that role).

Not a fan of the newtype solution as that would add quite a bit of boilerplate.

Reply:
I strongly dislike the practice performed by a handful of computer frameworks that auto-generate names by munging other names into a different case convention.  I would much rather have the literal string `DemoFooV1Marker`.

Reply:
In that case I think it'd be neater if the default marker name wasn't generated from the struct name but was also a required param.

And if we don't use equivalent names, should we at least verify that the key's version matches the struct's version?

Reply:
Good call.  I changed the macro so that the marker type is explicit.

> And if we don't use equivalent names, should we at least verify that the key's version matches the struct's version?

Seems unnecessary to me, but we could do it in a follow-up.